contracts/DisputableDelay.sol

```diff
@@ -0,0 +1,264 @@
+Executable (using canExecute())
+Cancelled (by Agreements app)
+Executed
+*/
+
+contract DisputableDelay is DisputableAragonApp, IForwarder {
+    using SafeMath64 for uint64;
+
+    /**
+        bytes32 public constant SET_DELAY_ROLE = keccak256("SET_DELAY_ROLE");
+        bytes32 public constant DELAY_EXECUTION_ROLE = keccak256("DELAY_EXECUTION_ROLE");
+    */
+    bytes32 public constant SET_DELAY_ROLE = 0x2b56821903fce674a357ec19f2dd583428c0cc74e6be302c348b3a3ffa5b4f66;
+    bytes32 public constant DELAY_EXECUTION_ROLE = 0x68a7ca9b0904e026378fb888f3be95ada2a0c0b11f58c40530c1a383c0f99ce9;
+
+    string private constant ERROR_NO_SCRIPT = "DELAY_NO_SCRIPT";
+    string private constant ERROR_CANNOT_FORWARD = "DELAY_CANNOT_FORWARD";
+    string private constant ERROR_CANNOT_EXECUTE = "DELAY_CANNOT_EXECUTE";
+    string private constant ERROR_CANNOT_CLOSE = "DELAY_CANNOT_CLOSE";
+
+    enum DelayedScriptStatus {
+        Active,              // A delayed script that has been reported to Agreements
+        Paused,              // A delayed script that is being challenged by Agreements
+        Cancelled,           // A delayed script that has been cancelled by Agreements
+        Executed             // A delayed script that has been executed
+    }
+
+    struct DelayedScript {
+        uint64 executionFromTime;
+        uint64 pausedAt;
+        DelayedScriptStatus delayedScriptStatus;
+        bytes evmCallScript;
+        uint256 actionId;
+    }
+
+    uint64 public executionDelay;
+    uint256 public delayedScriptsNewIndex = 0;
+    mapping (uint256 => DelayedScript) public delayedScripts;
+
+    event ExecutionDelaySet(uint64 indexed executionDelay);
+    event DelayedScriptStored(uint256 indexed delayedScriptId, uint256 indexed actionId, bytes evmScript);
+    event ExecutionPaused(uint256 indexed delayedScriptId, uint256 indexed actionId);
+    event ExecutionResumed(uint256 indexed delayedScriptId, uint256 indexed actionId);
+    event ExecutionCancelled(uint256 indexed delayedScriptId, uint256 indexed actionId);
+    event AgreementActionClosed(uint256 indexed delayedScriptId, uint256 indexed actionId);
+    event ExecutedScript(uint256 indexed delayedScriptId, uint256 indexed actionId);
+
+    /**
+    * @notice Initialize the Delay app
+    * @param _executionDelay The delay in seconds a user will have to wait before executing a script
+    */
+    function initialize(uint64 _executionDelay) external onlyInit {
+        initialized();
+        executionDelay = _executionDelay;
+    }
+
+    /**
+    * @notice Set the execution delay to `_executionDelay`
+    * @param _executionDelay The new execution delay
+    */
+    function setExecutionDelay(uint64 _executionDelay) external auth(SET_DELAY_ROLE) {
+        executionDelay = _executionDelay;
+
+        emit ExecutionDelaySet(executionDelay);
+    }
+
+    /**
+    * @dev IDisputable interface conformance
+    */
+    function getDisputableAction(uint256 _delayedScriptId)
```

Comment:
This one has been removed for now from the interface, see [this discussion](https://github.com/aragon/aragonOS/pull/581#discussion_r439386331) for reference.